### DISCUSSION TITLE
Flytesandbox wont start up - Error 137 on debian 11

### DISCUSSION THREAD

**Discussion by jinserk** [flyteorg/flyte#1141] [Q&A] [answered]
Hi,

I'm totally newbie in `flyte` and trying to check if it will meet our use from the scratch.
However, when I tried to `make start` in the `flytesnacks` I have an error like:
```
$ make start
Starting Flyte sandbox
Waiting for Flyte to become ready...
make: *** [Makefile:65: wait] Error 137
```

I have my local debian bullseye distro with `kubectl` and `minikube` as:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:18:45Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7", GitCommit:"132a687512d7fb058d0f5890f07d4121b3f0a2e2", GitTreeState:"clean", BuildDate:"2021-05-12T12:32:49Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}

$ minikube version
minikube version: v1.21.0
commit: 76d74191d82c47883dc7e1319ef7cebd3e00ee11
```

What should I do more to see the console and test the flytekit?

**Comment by kumare3** [comment]
@jinserk `Error 137` usually means that your docker daemon does not have enough memory associated with it. 
So `make start` actually starts a `docker container` in the back. but, it seems the docker container failed because of memory pressure.
Can you restart docker daemon with more memory?

**Reply by jinserk** [reply]
Thank you @kumare3 for answering. From google search, I checked the memory limit with the command `docker container stats` and the result is:
```
CONTAINER ID   NAME       CPU %     MEM USAGE / LIMIT     MEM %     NET I/O          BLOCK I/O       PIDS
8c8b164f8a41   minikube   24.39%    835.5MiB / 15.34GiB   5.32%     564kB / 3.35MB   987kB / 311MB   433
```

I'm the docker newbie too so don't know how to increase the mem limit. Can you help me a little bit more?
My laptop has 32GB main memory with 8 core CPUs. Isn't it sufficient to run the `flyte`?

**Reply by kumare3** [reply]
Ohh that's more than enough to run Flyte, but the thing is we run kubernetes inside docker and run Flyte in it. Can you tell me the operating system you are using and I can help. Also join the slack channel and I or one of the community members can help and get you started ASAP https://docs.google.com/forms/d/e/1FAIpQLScWPJZncyL-bBCWbuQ9HWuMzA8-r2RJc__CSWJoAYUQADN-BQ/viewform

**Reply by jinserk** [reply]
As I wrote on the question above, my laptop OS is Debian 11 (Bullseye), the testing version.
```
$ uname -a
Linux jupiter 5.10.0-7-amd64 #1 SMP Debian 5.10.40-1 (2021-05-28) x86_64 GNU/Linux
```
I registered to join the slack. Thank you for your help again!

**Comment by jinserk** [comment]
I've got a comment from https://github.com/k3s-io/k3s/issues/3500#issuecomment-867032915.
Could you figure it out the `cgroups` thing in the Flytesnack container?

**Reply by kumare3** [reply]
Interesting we are looking into this

**Comment by kumare3** [comment, marked as answer]
@jinserk thank you for the issue, we did find that `moby` changed to add a special entrypoint that would enable some flags for cgroups v2. We have added support for that in flyte-sandbox also - https://github.com/flyteorg/flyte/pull/1175
